query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/cache/CacheSessionProvider.scala

```diff
@@ -1,7 +1,36 @@
 package com.socrata.querycoordinator.caching.cache
 
 import com.rojoma.simplearm.v2.ResourceScope
+import com.socrata.soql.SoQLAnalysis
+import com.socrata.soql.types.SoQLType
+import org.joda.time.DateTime
+import org.slf4j.Logger
 
 trait CacheSessionProvider {
-  def open(rs: ResourceScope): CacheSession
+  protected val log: Logger
+
+  def open(rs: ResourceScope, dataset: String): CacheSession
+
+  private var disabledTil = new DateTime(0)
+
+  def disabled() = {
```

Comment:
This looks like a function that returns whether or not the cache provider is disabled; I'd prefer something more imperative-sounding, like "disable"